### PR TITLE
[WIP] Meta lamp/lending pool/nix configuration

### PR DESCRIPTION
**Pull request by olgaklimenko**
For setting up nix, I used docs, presented in [haskell.nix repo](https://input-output-hk.github.io/haskell.nix/)

Niv https://github.com/nmattia/niv is used to add plutus dependency, you can use it to add new dependency to the `source.json`. As I understood, repo in dependencies should contain `default.nix` file. Niv is recommended [here](https://input-output-hk.github.io/haskell.nix/tutorials/getting-started/#setting-up-the-binary-cache) to setup the binary hash. 

Other deps can be added to cabal.project with a [source repository hash](https://input-output-hk.github.io/haskell.nix/tutorials/source-repository-hashes/).

### CODE REVIEW

File: MetaLamp/lending-pool/client/package.json [hunk 1]
```diff
@@ -26,8 +26,10 @@
     "jquery": "^3.3.1",
     "json-bigint": "^1.0.0",
     "node-sass": "^4.12.0",
+    "purescript": "^0.13.6",
```
Comment (performanceArtist) [inline MetaLamp/lending-pool/client/package.json]: Is there a need to specify `purescript` and `spago` as local dependencies? Aren't they available via nix-shell already? I think this causes `npm install` to build `spago`, `purescript` and `ghc`. Also it fails for me.

Screenshot from 2021-06-28 16-16-12

Comment (performanceArtist) [inline MetaLamp/lending-pool/client/package.json]: Actually if I remove these dependencies, everything seems to work.

Comment (olgaklimenko) [reply MetaLamp/lending-pool/client/package.json]: @stanislav-az can you please try to test client too without `purescript` and `spago` in `package.json`? Does it work for you?

Comment (stanislav-az) [reply MetaLamp/lending-pool/client/package.json]: @olgaklimenko it works for me without modifications to `package.json` (the current version in PR is working)

Comment (stanislav-az) [reply MetaLamp/lending-pool/client/package.json]: @olgaklimenko removing `purescript` and `spago` from `package.json` works too though.

**Review by stanislav-az — DISMISSED** [review]

**Comment by stanislav-az** [comment]
In current versions the local versions of cabal purs and spago are used. We need to fix that:
```
[nix-shell:~/iohk/plutus-use-cases/MetaLamp/lending-pool]$ which cabal 
/usr/bin/cabal

[nix-shell:~/iohk/plutus-use-cases/MetaLamp/lending-pool]$ which spago
/usr/local/bin/spago

[nix-shell:~/iohk/plutus-use-cases/MetaLamp/lending-pool]$ which purs
/usr/bin/purs
```
For comparison here is the desired behavior from plutus repo nix shell:
```
[nix-shell:~/iohk/plutus]$ which purs 
/nix/store/qj2cifdaarm8akqk4qq4sr3spar5ra0b-purescript-v0.13.8/bin/purs

[nix-shell:~/iohk/plutus]$ which spago 
/nix/store/qcxkhwwgr1adpniwprb52wij6pby1xp6-spago-0.19.1/bin/spago

[nix-shell:~/iohk/plutus]$ which cabal 
/nix/store/i6d90ab5v977xfqqljnijs2nlfqardhw-cabal-install-exe-cabal-3.4.0.0/bin/cabal
```

**Comment by olgaklimenko** [comment]
Close PR, because have newer nix configuration based on plutus-starter nix config in [PR](https://github.com/input-output-hk/plutus-use-cases/pull/60)

**Pull request closed without merging**